### PR TITLE
Migrate Elasticsearch tests to use BaseConnectorTest

### PR DESCRIPTION
**Pull request by findinpath**
This PR addresses https://github.com/trinodb/trino/issues/7109

Overview of changes:
 
- BaseElasticsearchSmokeTest.java has been renamed to BaseTestElasticsearchConnectorTest.java
- TestElasticsearch6IntegrationSmokeTest.java has been renamed to TestElasticsearch6ConnectorTest.java
- TestElasticsearch7IntegrationSmokeTest.java has been renamed to TestElasticsearch7ConnectorTest.java
- Added custom implementations in BaseTestElasticsearchConnectorTest for the tests that don't match the generic expectations.

### CODE REVIEW

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseTestElasticsearchConnectorTest.java [hunk 1]
```diff
@@ -39,22 +41,22 @@
...
-        extends AbstractTestIntegrationSmokeTest
+public abstract class BaseTestElasticsearchConnectorTest
+        extends BaseConnectorTest
 {
     private final String image;
     private ElasticsearchServer elasticsearch;
     protected RestHighLevelClient client;
 
-    BaseElasticsearchSmokeTest(String image)
+    BaseTestElasticsearchConnectorTest(String image)
```
Comment (hashhar) [inline plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseTestElasticsearchConnectorTest.java:59]: nit: existing convention is `BaseElasticsearchConnectorTest` for the abstract base classes. See `BaseSqlServerConnectorTest`.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseTestElasticsearchConnectorTest.java [hunk 2]
```diff
@@ -106,6 +155,68 @@ public void testDescribeTable()
         assertEquals(actualColumns, expectedColumns);
     }
 
+    @Test
+    @Override
+    public void testPredicateReflectedInExplain()
+    {
+        // The format of the string representation of what gets shown in the table scan is connector-specific
+        // and there's no requirement that the conform to a specific shape or contain certain keywords.
```
Comment (hashhar) [inline plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseTestElasticsearchConnectorTest.java:163]: While this is true I think as an end-user it makes sense to make them consistent where possible. After all it's part  of the explain output.

WDYT @findepi @martint ?
We can address this in a follow-up if needed.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseTestElasticsearchConnectorTest.java [hunk 3]
```diff
@@ -106,6 +155,68 @@ public void testDescribeTable()
...
+        assertExplain(
+                "EXPLAIN SELECT name FROM nation ORDER BY nationkey DESC NULLS LAST LIMIT 5",
+                "TopNPartial\\[5 by \\(nationkey DESC");
+    }
+
+    @Test
+    @Override
+    public void testLimitPushDown()
```
Comment (hashhar) [inline plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseTestElasticsearchConnectorTest.java:183]: I strongly feel about renaming the test in ATDQ to `testLimitQueries` - it doesn't test pushdown at all and most subclasses already have another test called `testLimitPushdown` (vs `testLimitPushDown`) which is very confusing/error-prone. cc: @losipiuk
I can do this as a part of https://github.com/trinodb/trino/pull/7495.

No change requested here.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseTestElasticsearchConnectorTest.java [hunk 4]
```diff
@@ -960,6 +1091,22 @@ public void testMissingIndex()
         assertTableDoesNotExist("nonexistent_table");
     }
 
+    @Override
+    protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
```
Comment (hashhar) [inline plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseTestElasticsearchConnectorTest.java:1095]: nit: Move to top of the class.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseTestElasticsearchConnectorTest.java [hunk 5]
```diff
@@ -969,6 +1116,27 @@ protected void assertTableDoesNotExist(String name)
...
+     *
+     * @return the amount of clauses to be used in large queries
+     */
+    @Override
+    protected Object[][] largeInValuesCountData()
+    {
+        return new Object[][] {
+                {200},
+                {500},
+                {1000}
```
Comment (hashhar) [inline plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseTestElasticsearchConnectorTest.java:1136]: nit: Maybe use 1024 instead of 1000 since that's the max supported?

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java [hunk 6]
```diff
@@ -23,10 +23,10 @@
 
 import static java.lang.String.format;
 
-public class TestElasticsearch6IntegrationSmokeTest
```
Comment (hashhar) [inline plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java:26]: @findepi Do you think it should be enough to use SmokeTest for the different variants similar to how we do it for MemSQL etc.

@findinpath I think you should instead change the ES7 tests to extend `BaseConnectorSmokeTest`. The smoke test is a light handed test which tests most functionality at a high level while the `BaseConnectorTest` is very thorough and long running.

File: testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueries.java [hunk 7]
```diff
@@ -325,8 +330,8 @@ public void testShowColumns()
                 .build();
 
         // Until we migrate all connectors to parametrized varchar we check two options
-        assertTrue(actual.equals(expectedParametrizedVarchar) || actual.equals(expectedUnparametrizedVarchar),
-                format("%s does not matche neither of %s and %s", actual, expectedParametrizedVarchar, expectedUnparametrizedVarchar));
+        assertTrue(actual.equals(expectedParameterizedVarchar) || actual.equals(expectedUnparameterizedVarchar),
+                format("%s does not match neither of %s and %s", actual, expectedParameterizedVarchar, expectedUnparameterizedVarchar));
```
Comment (hashhar) [inline testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueries.java:334]: nit: commit message can be a more general `fix typos`.

**Review by hashhar — COMMENTED** [review]
Thanks a lot for doing this. ❤️ 

Should be possible to remove most commented tests by adjusting `hasBehaviour` declaration.
For the ones which cannot use a `SkipException` instead of plain comment.

Also some comments about smoke tests vs full test for ES7.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java (hunk 6 above)
Comment (findinpath) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java:26]: @hashhar  the requirement stated in https://github.com/trinodb/trino/issues/7109 is about replacing the smoke test class with  the thorough `BaseConnectorTest` classes. I've followed the requirements of the issue specified above (cc: @losipiuk )

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseTestElasticsearchConnectorTest.java (hunk 1 above)
Comment (findinpath) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseTestElasticsearchConnectorTest.java:59]: @hashhar  I've followed here the requirements from https://github.com/trinodb/trino/issues/7109 . I find though you suggestion good and I'll do the modification. Thanks.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseTestElasticsearchConnectorTest.java (hunk 5 above)
Comment (findinpath) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseTestElasticsearchConnectorTest.java:1136]: I've tried with `1024` and the test fails ... I can put `1023` , but I assume that it doesn't make really a difference in the test to test the borders of the default Elasticsearch setting.

Comment (hashhar) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseTestElasticsearchConnectorTest.java]: @losipiuk I'm not sure if I remember correctly but don't connectors indicate to the engine how large in-clauses they can handle? Was it something like domain compaction threshold? Should Elasticsearch connector also be setting it?

Comment (losipiuk) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseTestElasticsearchConnectorTest.java]: There is no general mechanism for that for what I see. There is configuration and logic in JDBC based connectors and Hive connector. We could add logic for ES to, but it is not like flipping a flag. A few lines of code would need to be added. Let's keep test refactor and that separate.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java (hunk 6 above)
Comment (losipiuk) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java:26]: When issue was created we did not have `BaseConnectorSmokeTest` yet :) What @hashhar says makes total sense to me.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseTestElasticsearchConnectorTest.java (hunk 3 above)
Comment (losipiuk) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseTestElasticsearchConnectorTest.java]: Should we just merge with `testLimit`?

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java [hunk 8]
```diff
@@ -79,6 +131,20 @@ public final void destroy()
         client.close();
     }
 
+    @Test
+    @Override
+    public void testCreateSchema()
+    {
+        throw new SkipException("Schema creation is not supported in the Elasticsearch connector");
```
Comment (losipiuk) [inline plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java:138]: It would be nice to add
```
assertThatThrown(() -> super.testCreateSchema()).
 .hasMessageContaining("....")
```
To validate that reason for original test failure is actually fact that we cannot create schema. 
It will remind us that test need to be changed when (if) we add support for schema creation to ES.

File: testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueries.java [hunk 9]
```diff
@@ -300,7 +300,7 @@ public void testShowColumns()
     {
         MaterializedResult actual = computeActual("SHOW COLUMNS FROM orders");
 
-        MaterializedResult expectedUnparametrizedVarchar = resultBuilder(getSession(), VARCHAR, VARCHAR, VARCHAR, VARCHAR)
+        MaterializedResult expectedUnparameterizedVarchar = resultBuilder(getSession(), VARCHAR, VARCHAR, VARCHAR, VARCHAR)
```
Comment (losipiuk) [inline testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueries.java:308]: Commit message `fix typo` -> `Fix typo`

**Review by losipiuk — COMMENTED** [review]
Looks nice. Thanks

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseTestElasticsearchConnectorTest.java (hunk 2 above)
Comment (findepi) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseTestElasticsearchConnectorTest.java]: Yes, consistency is good to have. 
The problem is that ES TH doesn't reveal associated `constraint` at all. If this gets left for a follow up, let's have a TODO with an issue

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java (hunk 6 above)
Comment (findepi) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java:26]: > @findepi Do you think it should be enough to use SmokeTest for the different variants similar to how we do it for MemSQL etc.

totally!

**Review by hashhar — COMMENTED** [review]
Looks good. Some comments.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java [hunk 10]
```diff
@@ -79,6 +131,20 @@ public final void destroy()
...
+    }
+
+    @Test
+    @Override
+    public void testRenameTableAcrossSchema()
```
[the remark this answers is not in the document]
Comment (findinpath) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java:143]: I've added `SUPPORTS_RENAME_TABLE_ACROSS_SCHEMAS` and `SUPPORTS_RENAME_TABLE` as `false` in `hasBehaviour`.

ElasticsearchMetadata doesn't have an implementation for `io.trino.spi.connector.ConnectorMetadata#renameTable`

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java [hunk 11]
```diff
@@ -106,6 +176,68 @@ public void testDescribeTable()
         assertEquals(actualColumns, expectedColumns);
     }
 
+    @Test
+    @Override
+    public void testPredicateReflectedInExplain()
+    {
+        // The format of the string representation of what gets shown in the table scan is connector-specific
+        // and there's no requirement that the conform to a specific shape or contain certain keywords.
```
[the remark this answers is not in the document]
Comment (findinpath) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java:184]: I need a little help regarding the `TODO` because I don't know the convention regarding such things in the project.
Can you describe exactly what do you expect me to do here?

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java (hunk 6 above)
Comment (findinpath) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java:26]: `TestElasticsearch7ConnectorSmokeTest` extends now `BaseConnectorSmokeTest`

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java (hunk 11 above)
[the remark this answers is not in the document]
Comment (hashhar) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java:179]: @findinpath When adding a TODO in the code we generally create an issue in GitHub describing what needs to be done to be able to remove the TODO. And then in the TODO comment we add a link to the GitHub issue - that way future readers (or the person closing the issue) can update the code and remove TODO comments.

So you can just create a GitHub issue like `Provide consistent representation of pushed-down operations in EXPLAIN output` (and maybe add a link to https://github.com/trinodb/trino/pull/7609#discussion_r616148357).

[the remark this answers is not in the document]
Comment (findinpath) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java:179]: I hope I got it right.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java [hunk 12]
```diff
@@ -23,10 +23,10 @@
...
-        extends BaseElasticsearchSmokeTest
+public class TestElasticsearch6ConnectorTest
```
Comment (losipiuk) [inline plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java:26]: @hashhar one question here. You suggested that `BaseConnectorTest` is used for testing ES6 while `BaseConnectorSmokeTest` for ES7. Not sure if that matters but maybe we should test newest version with more verbose `BaseConnectorTest`?

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java [hunk 13]
```diff
@@ -39,26 +41,55 @@
...
 import static java.lang.String.format;
 import static java.nio.charset.StandardCharsets.UTF_8;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.testng.Assert.assertFalse;
 import static org.testng.Assert.assertTrue;
 
-public abstract class BaseElasticsearchSmokeTest
-        // TODO extend BaseConnectorTest
-        extends AbstractTestIntegrationSmokeTest
+public abstract class BaseElasticsearchConnectorTest
```
Comment (losipiuk) [inline plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java:52]: We probably can drop this one now. Given fact that we have just one subclass.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java [hunk 14]
```diff
@@ -106,6 +171,58 @@ public void testDescribeTable()
         assertEquals(actualColumns, expectedColumns);
     }
 
+    @Test
+    @Override
+    public void testPredicateReflectedInExplain()
+    {
+        // The format of the string representation of what gets shown in the table scan is connector-specific
+        // and there's no requirement that the conform to a specific shape or contain certain keywords.
+        //TODO (https://github.com/trinodb/trino/issues/7673) Provide consistent representation of pushed-down operations in EXPLAIN output
```
Comment (losipiuk) [inline plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java:180]: Thanks @findinpath . I reworded a bit.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java (hunk 12 above)
Comment (hashhar) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java:26]: I suggested based on how Cassandra/MemSQL/Mongo are tested today. I think the older ones require deeper testing since the upstream projects generally don't break things in newer versions (while something working in new version doesn't guarantee it worked in older version).

I don't have a strong opinion - very less information to make an informed decision.

Comment (findinpath) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java:26]: Elasticsearch 7 has been around for quite some time now  (more than 2 years actually)

https://github.com/elastic/elasticsearch/releases

Elasticsearch 6 has the latest bugfix release made about 5 months ago.
Elasticsearch 7 has the latest minor release made a month ago.

I tend to say that Elasticsearch 7 is more relevant at the time of this writing than Elasticsearch 6.

Comment (losipiuk) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java:26]: > Elasticsearch 7 has been around for quite some time now (more than 2 years actually)
> 
> https://github.com/elastic/elasticsearch/releases
> 
> Elasticsearch 6 has the latest bugfix release made about 5 months ago.
> Elasticsearch 7 has the latest minor release made a month ago.
> 
> I tend to say that Elasticsearch 7 is more relevant at the time of this writing than Elasticsearch 6.

I would go with testing more the version is more broadly used. I have now data on that at all. But from what I read above it seems like ES7 is the one to have broader test coverage. @findepi @martint opinion?

Comment (findepi) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java:26]: @losipiuk see discussion under https://github.com/trinodb/trino/issues/5804. Does this help?

Comment (findinpath) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java:26]: https://github.com/trinodb/trino/issues/5804#issuecomment-721365549

Maybe Elasticsearch 6 & Elasticsearch 7 full tested and a Elasticsearch 7 latest smoke test ? 

What do you think about this?

Comment (losipiuk) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java:26]: It is safe for sure. But unless there are significant differences between 6 and 7. And both are commonly used I would rather have just one full tests, to keep CI runs as short as possible.

Comment (findinpath) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java:26]: Good. Then I'll keep things as they are now.

I'll just drop then the base class `BaseElasticsearchConnectorTest` and include everything in `TestElasticsearch6ConnectorTest.java ` .

Comment (martint) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java:26]: There are some significant differences between 6 and 7. Major versions of ES tend to make breaking changes in their APIs. That’s why we even added tests for both at all.

Comment (losipiuk) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java:26]: Thanks for chiming in. Still smoke test should be enough I think. The concept of smoke test class is that it still provides good test coverage; each important code path should be tested; we are just passing on testing every tiny corner case for each code path. So IMO it is reasonable trade-off to have just one full test class, and gain on faster CI trunaround.

cc: @findepi

**Review by hashhar — APPROVED** [review]
Thanks. LGTM.

**Comment by martint** [comment]
I haven't looked closely, but we seem to have lost all the ES-specific tests in TestElasticsearch7ConnectorSmokeTest that were in the former BaseElasticsearchSmokeTest. Some of those tests are important, as there are differences in how ES 6 and ES 7 handle index mappings (the shape of the urls, contents of the documents, etc).

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java (hunk 12 above)
Comment (martint) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java:52]: The issue is that in this change we've trade off testing ES-specific behaviors for generic connector behaviors in the ES 7 smoke test. I don't think we should do that -- those tests were there for a reason, as ES 6 and 7 have differences in how they expose index mappings (the shape of the URL and contents). 

I'm ok with not running the full suite of queries, but the tests that currently exist exercise certain areas that are not covered by any of the generic smoke tests, and we should remove them (e.g, nested documents, all supported ES data types, coercions from ES types, pushdown on specific types, index aliases, etc)

**Review by martint — CHANGES_REQUESTED** [review]

**Comment by findinpath** [comment]
@losipiuk  & @martint it would be helpful here to get an understanding of what needs to be done here concerning Elaticsearch 7 tests.

If this class TestElasticsearch7ConnectorSmokeTest is supposed to be a smoke test class what new test cases should it include?

I need a few specific pointers to get this PR accepted by all the parties involved in the PR.

Thank you all for the feedback received so far.

**Comment by losipiuk** [comment]
> @losipiuk & @martint it would be helpful here to get an understanding of what needs to be done here concerning Elaticsearch 7 tests.
> 
> If this class TestElasticsearch7ConnectorSmokeTest is supposed to be a smoke test class what new test cases should it include?
> 
> I need a few specific pointers to get this PR accepted by all the parties involved in the PR.
> 
> Thank you all for the feedback received so far.

If @martint feels strongly about necessity to test both ES6 and ES7 at same level of detail let's get back to version when you had both: 
* TestElasticsearch6ConnectorTest
* TestElasticsearch7ConnectorTest
based on `BaseElasticsearchConnectorTest` which is based on `BaseConnectorTest`

Sorry for back and forth on this one. I appreciate the effort a lot.

**Comment by martint** [comment]
> If @martint feels strongly about necessity to test both ES6 and ES7 at same level of detail

It's ok not to run the full battery of tests, but the smoke tests should include the specific tests that were designed for ES. So, for instance, it doesn't matter whether `BaseConnectorTest.testAggregation` runs on both, as that's mostly a set of engine-level tests, but it does matter that one like `BaseElasticsearchSmokeTest.testArrayFields` do.

**Comment by findinpath** [comment]
@martint I'd appreciate having specific requirements concerning  `TestElasticsearch7ConnectorSmokeTest` class.

I'm not sure whether I understand correctly your comment: 

> It's ok not to run the full battery of tests, but the smoke tests should include the specific tests that were designed for ES. So, for instance, it doesn't matter whether BaseConnectorTest.testAggregation runs on both, as that's mostly a set of engine-level tests, but it does matter that one like BaseElasticsearchSmokeTest.testArrayFields or testDataTypes do.



Do you imply duplicating some of the methods from the class `TestElasticsearch6ConnectorTest` ? If yes, please point out which methods fall in this situation.

**Comment by hashhar** [comment]
@findinpath I think Martin means all tests from `BaseElasticsearchSmokeTest` (before changes) must also run against ES7.

The easiest way to do it would be what Lukasz suggested in https://github.com/trinodb/trino/pull/7609#issuecomment-827171218. (make use of the BaseElasticsearchConnectorTest to hold those ES specific tests and inherit them for both ES6 and ES7).

**Comment by findinpath** [comment]
@hashhar  this would mean going back to `TestElasticsearch7ConnectorTest` from `TestElasticsearch7ConnectorSmokeTest` .

So no more smoke tests for ES7.
I'm still confused :)

**Review by losipiuk — APPROVED** [review]

**Comment by losipiuk** [comment]
@martint can you please drop the `changes requested flag` if it looks fine to you now?

**Review by martint — APPROVED** [review]
Thanks @findinpath, and sorry for the confusion and back-and-forth. I was just trying to make sure we don't lose some important tests.

**Comment by findinpath** [comment]
No worries guys. I am thankful for being given the opportunity to contribute to this project.
